ciphers/VigenereCipher.java

```diff
@@ -0,0 +1,41 @@
+/*
+Author: Nathan Fei
+
+An implementation of the Vigenère cipher
+*/
+
+import java.util.Scanner;
+
+public class VigenereCipher {
+	public static void main(String[] args) {
+		Scanner sc = new Scanner(System.in);
+		System.out.print("Enter Message (letters only): ");
+		String message = sc.nextLine().toUpperCase();
+		System.out.print("Enter Key: ");
+		String key = sc.nextLine().toUpperCase();
+		System.out.print("[E]ncode or [d]ecode message?: ");
+		char choice = sc.next().toUpperCase().charAt(0);
+		if (choice == 'E') System.out.println(encode(message, key));
+		else System.out.println(decode(message, key));
+	}
+	
+	private static String encode(String message, String key) {
+		StringBuilder sb = new StringBuilder(message.length());
+		int keyPlace = 0;
+		for (char c : message.toCharArray()) {
+			sb.append((char)((c - 'A' + key.charAt(keyPlace) - 'A') % 26 + 'A'));
```

Comment:
I assume they are using StringBuilder in this instance since they are doing multiple operations and additions to the String. This should be more efficient. The JVM will use a StringBuilder each time there is an operation like that anyways, except it will create a new instance of StringBuilder for each separate operation. In this case, using just one instance will be more efficient. Of course, I am not sure if this is their reasoning, but just thought I would share.